### DISCUSSION TITLE
are port numbers no longer supported in Version 2.0?

### DISCUSSION THREAD

**Discussion by tritigr** [sqlalchemy/sqlalchemy#9984] [Usage Questions]
Hello, 
I used to be able to connect to my database from within jupyter like this:

db_url = f'postgresql+psycopg2://' + \
                creds['user_name'] + ':' + creds['password'] + '@' + \
                creds['host_name'] + ':' + creds['port_num'] + '/' + creds['db_name']
engine = sqlalchemy.create_engine(db_url)
%sql engine

With Versions 2.0 I get an error about a malformed connection string. If I leave out the port number, I get no error, but no connection, as I cannot use the standard port.

I didn't find anything about port numbers in the documentation..

Best regards
Til

**Comment by zzzeek** [comment]
there is no change in URL format and I dont see anything obviously wrong with the above URL.

There was a change in 1.4 however regarding `@` signs in passwords, so if you have special characters in passwords you want to make sure you [url escape any special characters](https://docs.sqlalchemy.org/en/20/core/engines.html#escaping-special-characters-such-as-signs-in-passwords) in the password (as well as the username also)

**Comment by tritigr** [comment]
Hi, thank you for your feedback.
There are no critical characters in the string, I think.
As I have no time to investigate this further, I'll stick to version 1.4 for the time being.

BR

**Reply by CaselIT** [reply]
as you prefer. as mike mentioned nothing has changed on the url logic between 1.4 and 2.0
